converter/dicom-cast/src/Microsoft.Health.DicomCast.TableStorage/Features/Storage/TableClientReadWriteTestProivder.cs

```diff
@@ -5,42 +5,41 @@
 
 using System.Threading;
 using System.Threading.Tasks;
+using Azure.Data.Tables;
 using EnsureThat;
-using Microsoft.Azure.Cosmos.Table;
-using Microsoft.Health.DicomCast.TableStorage.Configs;
 using Microsoft.Health.DicomCast.TableStorage.Features.Storage.Entities;
 
 namespace Microsoft.Health.DicomCast.TableStorage.Features.Storage
 {
-    public class TableClientReadWriteTestProivder : ITableClientTestProvider
+    public class TableClientReadWriteTestProvider : ITableClientTestProvider
     {
         private const string TestPartitionKey = "testpartition";
         private const string TestRowKey = "testrow";
         private const string TestData = "testdata";
         private const string TestTable = "testTable";
 
-        /// <inheritdoc/>
-        public async Task PerformTestAsync(CloudTableClient client, TableDataStoreConfiguration configuration, CancellationToken cancellationToken = default)
+        private readonly TableServiceClient _testServiceClient;
+
+        public TableClientReadWriteTestProvider(TableServiceClient testServiceClient)
         {
-            EnsureArg.IsNotNull(client, nameof(client));
-            EnsureArg.IsNotNull(configuration, nameof(configuration));
+            _testServiceClient = EnsureArg.IsNotNull(testServiceClient, nameof(testServiceClient));
+        }
 
-            CloudTable table = client.GetTableReference(TestTable);
-            await table.CreateIfNotExistsAsync(cancellationToken);
+        /// <inheritdoc/>
+        public async Task PerformTestAsync(CancellationToken cancellationToken = default)
+        {
+            await _testServiceClient.CreateTableIfNotExistsAsync(TestTable, cancellationToken: cancellationToken);
 
+            var tableClient = _testServiceClient.GetTableClient(TestTable);
             var entity = new HealthEntity(TestPartitionKey, TestRowKey) { Data = TestData };
 
-            var insertOrMergeOperation = TableOperation.InsertOrMerge(entity);
-
-            await table.ExecuteAsync(insertOrMergeOperation, cancellationToken);
+            await tableClient.UpsertEntityAsync(entity, cancellationToken: cancellationToken);
 
-            var retrieveOperation = TableOperation.Retrieve<HealthEntity>(TestPartitionKey, TestRowKey);
-            await table.ExecuteAsync(retrieveOperation, cancellationToken);
+            await tableClient.GetEntityAsync<HealthEntity>(TestPartitionKey, TestRowKey, cancellationToken: cancellationToken);
 
-            var deleteOperation = TableOperation.Delete(entity);
-            await table.ExecuteAsync(deleteOperation, cancellationToken);
+            await tableClient.DeleteEntityAsync(TestPartitionKey, TestRowKey, cancellationToken: cancellationToken);
 
-            await table.DeleteAsync(cancellationToken);
+            await _testServiceClient.DeleteTableAsync(TestTable, cancellationToken);
```

Comment:
Should we be wrapping the deletes in finally blocks? Or is that not important since we're upserting?

Reply:
I don't think it actually matters since we're doing both `CreateTableIfNotExistsAsync` and `UpsertEntityAsync`. This should mean that the method will complete regardless of the existing state of the table and entity.